exercise/cache/points.py

```diff
@@ -167,62 +228,78 @@ def r_augment(children: List[Dict[str, Any]]) -> None:
                     'unofficial': unofficial,
                     'date': submission.submission_time,
                     'url': submission.get_url('submission-plain'),
+                    'feedback_revealed': True,
+                    'feedback_reveal_time': None,
```

Comment:
Good point and yes, they are the same between all submissions. The main (and I think only) reason I decided to store these for every submission back when I implemented delayed feedback, was `points_badge`. `points_badge` is designed to work with three alternative input types (see `_points_data`): a `UserExerciseSummary` object, a `Submission` object or a CachedPoints dict. The last option does not differentiate between different types of cache entries - it may be an exercise, a submission, a module or a category. If we got rid of these values in the submission dict, the submission dropdowns on the course front page would not know if feedback is revealed or not. We would need extra logic that passes these values to `points_badge` along with the submission dict itself.

Note that these aren't the only values that are redundantly stored for each submission. `max_points` is also the same for all submissions. And this:
```python
'submission_count': 1, # to fool points badge
```

Reply:
That makes sense. Though, you have already added the `is_revealed` parameter to the `points_badge` template tag, but the parameter isn't always used (like in the course's Your points page). Let's not change this right now. We can leave this as is. It would be possible to avoid repeating these fields for every submission, but the changes would take too much time.